nixos/modules/security/pam.nix

```
@@ -370,156 +370,262 @@ let
       # Samba stuff to the Samba module.  This requires that the PAM
       # module provides the right hooks.
       text = mkDefault
-        (''
-          # Account management.
-          account required pam_unix.so
-          ${optionalString use_ldap
-              "account sufficient ${pam_ldap}/lib/security/pam_ldap.so"}
-          ${optionalString (config.services.sssd.enable && cfg.sssdStrictAccess==false)
-              "account sufficient ${pkgs.sssd}/lib/security/pam_sss.so"}
-          ${optionalString (config.services.sssd.enable && cfg.sssdStrictAccess)
-              "account [default=bad success=ok user_unknown=ignore] ${pkgs.sssd}/lib/security/pam_sss.so"}
-          ${optionalString config.krb5.enable
-              "account sufficient ${pam_krb5}/lib/security/pam_krb5.so"}
-          ${optionalString cfg.googleOsLoginAccountVerification ''
+        (
+          ''
+            # Account management.
+            account required pam_unix.so
+          '' +
+          optionalString use_ldap
+            ''
+              account sufficient ${pam_ldap}/lib/security/pam_ldap.so
+            '' +
+          optionalString (config.services.sssd.enable && cfg.sssdStrictAccess==false)
+            ''
+              account sufficient ${pkgs.sssd}/lib/security/pam_sss.so
+            '' +
+          optionalString (config.services.sssd.enable && cfg.sssdStrictAccess)
+            ''
+              account [default=bad success=ok user_unknown=ignore] ${pkgs.sssd}/lib/security/pam_sss.so
+            '' +
```

```suggestion
          optionalString (config.services.sssd.enable && cfg.sssdStrictAccess==false) ''
              account sufficient ${pkgs.sssd}/lib/security/pam_sss.so
          '' +
          optionalString (config.services.sssd.enable && cfg.sssdStrictAccess) ''
              account [default=bad success=ok user_unknown=ignore] ${pkgs.sssd}/lib/security/pam_sss.so
          '' +
```

like this it matches whats down below

---

Thanks! I've applied this suggestion to the whole thing.